### ISSUE TITLE
Allow setting request timeout

### ISSUE THREAD

**Issue by FRosner** [4n4nd/prometheus-api-client-python#277]
**Is your feature request related to a problem? Please describe.**<be>

We see requests getting stuck forever which blocks our app. The fix is to set a request timeout, which is not possible with the current APIs.

**Describe the solution you'd like**<br>

I'd like the ability to provide a global timeout when creating the client, as well as the ability to override the timeout for each request.

```
        response = self._session.get(
            "{0}/api/v1/query".format(self._url),
            params={**{"query": query}, **params},
            verify=self._ssl_verification,
            timeout=self._timeout,
        )
```

**Describe alternatives you've considered**<br>

I don't think there are good alternatives. You gotta be defensive towards requests getting stuck.

**Comment by FRosner** [comment]
@4n4nd let me know what you think about this. I can try to submit a PR, but I just wanted to confirm with you if you agree with the issue / proposed solution.

**Comment by 4n4nd** [comment]
yeah that seems like a reasonable solution. Could you please create a PR?
